### DISCUSSION TITLE
Using patched versions of `smithy-rs` crates

### DISCUSSION THREAD

**Discussion by spencergilbert** [awslabs/aws-sdk-rust#834] [Q&A] [answered]
👋 Hey there,

I've been trying to test a [proposed bug fix](https://github.com/awslabs/smithy-rs/pull/2567) for [an issue we ran into](https://github.com/awslabs/aws-sdk-rust/issues/749) and it's been difficult to actually _test_ the behavior of the proposed fix.

I've made the changes locally and updated our explicit dependency on `aws-smithy-http` to use my local crate via `path`, as well as using `patch.crates-io` to attempt to override the dependency for all of the other AWS crates we're pulling in.

I ended up apply the changes to a local copy of `aws-sdk-rust` on the `0.54.x` branch and using that as source of all our AWS dependencies.

Is that the best/recommended way or did I miss something easier?

**Comment by jdisanti** [comment, marked as answer]
The path you took sounds right to me.
